openapi_utils.go

```diff
@@ -188,37 +216,89 @@ func (oas *OpenAPISpec) PrepareOASRouter() *bunrouter.CompatRouter {
 }
 
 // FIXME: This is not a logic method of OAS, but could be a method of OASRouter
-func (oas *OpenAPISpec) FindPermission(OASRouter *bunrouter.CompatRouter, path string, method string) (XPermission, error) {
+func (oas *OpenAPISpec) FindPermission(OASRouter *bunrouter.CompatRouter, path string, method string) (RondConfig, error) {
 	recorder := httptest.NewRecorder()
 	responseReader := strings.NewReader("request-permissions")
 	request, _ := http.NewRequest(method, path, responseReader)
 	OASRouter.ServeHTTP(recorder, request)
 
 	if recorder.Code != http.StatusOK {
-		return XPermission{}, fmt.Errorf("%w: %s %s", ErrNotFoundOASDefinition, utils.SanitizeString(method), utils.SanitizeString(path))
+		return RondConfig{}, fmt.Errorf("%w: %s %s", ErrNotFoundOASDefinition, utils.SanitizeString(method), utils.SanitizeString(path))
 	}
 
 	recorderResult := recorder.Result()
 	rowFilterEnabled, err := strconv.ParseBool(recorderResult.Header.Get("resourceFilter.rowFilter.enabled"))
 	if err != nil {
-		return XPermission{}, fmt.Errorf("error while parsing rowFilter.enabled: %s", err)
+		return RondConfig{}, fmt.Errorf("error while parsing rowFilter.enabled: %s", err)
 	}
 	enableResourcePermissionsMapOptimization, err := strconv.ParseBool(recorderResult.Header.Get("options.enableResourcePermissionsMapOptimization"))
 	if err != nil {
-		return XPermission{}, fmt.Errorf("error while parsing rowFilter.enabled: %s", err)
+		return RondConfig{}, fmt.Errorf("error while parsing rowFilter.enabled: %s", err)
 	}
-	return XPermission{
-		AllowPermission: recorderResult.Header.Get("allow"),
-		ResponseFilter:  ResponseFilterConfiguration{Policy: recorderResult.Header.Get("responseFilter.policy")},
-		ResourceFilter: ResourceFilter{
-			RowFilter: RowFilterConfiguration{Enabled: rowFilterEnabled, HeaderKey: recorderResult.Header.Get("resourceFilter.rowFilter.headerKey")},
+	return RondConfig{
+		RequestFlow: RequestFlow{
+			PolicyName:    recorderResult.Header.Get("allow"),
+			GenerateQuery: rowFilterEnabled,
+			QueryOptions: QueryOptions{
+				HeaderName: recorderResult.Header.Get("resourceFilter.rowFilter.headerKey"),
+			},
+		},
+		ResponseFlow: ResponseFlow{
+			PolicyName: recorderResult.Header.Get("responseFilter.policy"),
 		},
 		Options: PermissionOptions{
 			EnableResourcePermissionsMapOptimization: enableResourcePermissionsMapOptimization,
 		},
 	}, nil
 }
 
+func newRondConfigFromPermissionV1(old *XPermission) *RondConfig {
```

Comment:
Here instead of old would be better to name the params v1Permission, or v1Config

Reply:
done!